chart/kubeapps/templates/kubeapps-frontend-config.yaml

```diff
@@ -57,36 +57,42 @@ data:
       # Ensure each cluster can be reached (should only be
       # used with an auth-proxy where k8s credentials never leave
       # the cluster). See clusters option.
-      {{- range .Values.clusters }}
+    {{- range .Values.clusters }}
       location ~* /api/clusters/{{ .name }} {
-        rewrite /api/clusters/{{ .name }}/(.*) /$1 break;
-        rewrite /api/clusters/{{ .name }} / break;
-
-      {{/* Helm returns a nil pointer error when accessing foo.bar if foo doesn't
+    {{/* We need to split the API service URL(s) into the base url and the path segment so
+           those configurations using a path can be appropriately rewritten below while
+           ensuring the proxy_pass statement is given the base URL only. */}}
+    {{- $parsed := urlParse (default "https://kubernetes.default" .apiServiceURL) }}
+    {{- $apiServiceBaseURL := urlJoin (pick $parsed "scheme" "host") }}
```

Comment:
@absoludity I changed the `printf` to `urlJoin (pick ...` here as you can see. Seems better to use a stdlib function to generate a [partial] URL instead of doing my own formatting 😄

Reply:
[`936ded6` (#2337)](https://github.com/kubeapps/kubeapps/pull/2337/commits/936ded61f610d1ab65efe7076184f0c962b1bdb5)

Reply:
Very nice!